apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/entity/blockchain/BlockImpl.java

```diff
@@ -380,69 +342,82 @@ public boolean checkSignature() {
     }
 
     @Override
-    public void setPrevious(Block block, HeightConfig config, Shard lastShard) {
+    public void setPrevious(Block[] threeLatestBlocksArray, HeightConfig config, Shard lastShard, int initialBlockHeight) {
```

Comment:
I think DTO instead of array (threeLatestBlocksArray) will make code more understandable.
For example: 
{
lastBlock,
secondBlock,
thirdBlock
}

Reply:
Here we need mostly two/three fields like: - timestamp, timeout, baseTarget
so we can optimize that code later after extracting consensus logic